roles/v2/pool/src/lib/mining_pool/mod.rs

```diff
@@ -81,28 +84,40 @@ impl Downstream {
         task::spawn(async move {
             debug!("Starting up downstream receiver");
             loop {
-                let receiver = cloned.safe_lock(|d| d.receiver.clone()).unwrap();
+                let receiver_res = cloned
```

Comment:
Can we get away with moving this outside the loop? Seems silly to clone it on every iteration right?